src/agent.ts

```diff
@@ -1,68 +1,43 @@
+import { Finding, HandleTransaction, TransactionEvent } from "forta-agent";
+import { CreateFinding } from "./findings";
 import {
-  BlockEvent,
-  Finding,
-  HandleBlock,
-  HandleTransaction,
-  TransactionEvent,
-  FindingSeverity,
-  FindingType,
-} from "forta-agent";
-
-export const ERC20_TRANSFER_EVENT =
-  "event Transfer(address indexed from, address indexed to, uint256 value)";
-export const TETHER_ADDRESS = "0xdAC17F958D2ee523a2206206994597C13D831ec7";
-export const TETHER_DECIMALS = 6;
-let findingsCount = 0;
-
-const handleTransaction: HandleTransaction = async (
-  txEvent: TransactionEvent
-) => {
-  const findings: Finding[] = [];
-
-  // limiting this agent to emit only 5 findings so that the alert feed is not spammed
-  if (findingsCount >= 5) return findings;
-
-  // filter the transaction logs for Tether transfer events
-  const tetherTransferEvents = txEvent.filterLog(
-    ERC20_TRANSFER_EVENT,
-    TETHER_ADDRESS
-  );
-
-  tetherTransferEvents.forEach((transferEvent) => {
-    // extract transfer event arguments
-    const { to, from, value } = transferEvent.args;
-    // shift decimals of transfer value
-    const normalizedValue = value.div(10 ** TETHER_DECIMALS);
-
-    // if more than 10,000 Tether were transferred, report it
-    if (normalizedValue.gt(10000)) {
-      findings.push(
-        Finding.fromObject({
-          name: "High Tether Transfer",
-          description: `High amount of USDT transferred: ${normalizedValue}`,
-          alertId: "FORTA-1",
-          severity: FindingSeverity.Low,
-          type: FindingType.Info,
-          metadata: {
-            to,
-            from,
-          },
-        })
-      );
-      findingsCount++;
+  FORTA_DEPLOYER_ADDRESS,
+  CREATE_AGENT_FUNCTION,
+  PROXY_CONTRACT_ADDRESS,
+} from "./constants";
+
+export function provideHandleTransaction(
+  deployerAddress: string,
+  createAgentFunction: string,
+  proxyContractAddress: string
+): HandleTransaction {
+  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+
+    // if the Transaction is not from Forta deployer address, return findings
+    if (txEvent.from !== deployerAddress.toLowerCase()) {
+      return findings;
     }
-  });
-
-  return findings;
-};
-
-// const handleBlock: HandleBlock = async (blockEvent: BlockEvent) => {
-//   const findings: Finding[] = [];
-//   // detect some block condition
-//   return findings;
-// }
-
+    // Filter the transaction for functions that match the create botfunction and that are sent to the proxy contract
+    const newAgents = txEvent.filterFunction(
```

Comment:
```suggestion
    const newBots = txEvent.filterFunction(
```

(The new term should be preferred)